### PR TITLE
Add verbose flag to ninja build.

### PR DESCRIPTION
**Pull request by jrhemstad**
## Description

Adds the `-v` option to the ninja build to enable the verbose compiler invocation for every object file.

This is useful for ensuring the correct arguments are passed to the compiler. 

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/build_common.sh [hunk 1]
```diff
@@ -101,7 +101,7 @@ function configure(){
 function build(){
     local BUILD_NAME=$1
     source "./sccache_stats.sh" start
-    cmake --build $BUILD_DIR --parallel $PARALLEL_LEVEL
+    cmake --build $BUILD_DIR --parallel $PARALLEL_LEVEL -- -v
```
Comment (robertmaynard) [inline ci/build_common.sh:104]:
```suggestion
    cmake --build $BUILD_DIR --parallel $PARALLEL_LEVEL -v
```

File: ci/windows/build_common.psm1 [hunk 2]
```diff
@@ -85,7 +85,7 @@ function build {
 
     sccache_stats('Start')
 
-    cmake --build $script:BUILD_DIR --parallel $script:PARALLEL_LEVEL
+    cmake --build $script:BUILD_DIR --parallel $script:PARALLEL_LEVEL -- -v
```
Comment (robertmaynard) [inline ci/windows/build_common.psm1:88]:
```suggestion
    cmake --build $script:BUILD_DIR --parallel $script:PARALLEL_LEVEL -v
```

**Review by robertmaynard — CHANGES_REQUESTED** [review]
cmake --build has a `-v` option which maps to the underlying generator command ( -v, VERBOSE=1, etc ).

**Review by miscco — APPROVED** [review]
looks good with @robertmaynard changes incorporated

**Review by robertmaynard — APPROVED** [review]

**Pull request merged**